openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -458,11 +460,15 @@ def get_object(self):
             username = self.request.GET.get('username', '') or self.request.user.username
         else:
             username = self.request.user.username
+        course_key = CourseKey.from_string(self.kwargs['course_key_string'])
         overview = CoursewareMeta(
-            CourseKey.from_string(self.kwargs['course_key_string']),
+            course_key,
             self.request,
             username=username,
         )
+        # Record course goals user activity for learning mfe courseware on web
+        if RECORD_USER_ACTIVITY_FLAG.is_enabled():
```

Comment:
I still maintain that this is bad API design, to push context like "what are the conditions in which this method should be skipped" onto callers of the API. I'm only harping on this because I think it's a mistake that happens a lot in the platform - where business logic bleeds outside of the abstractions meant to hide it. This is just a nit - I understand that the flag is temporary, so this is not a big deal.

Reply:
i do agree but i think once this is tested i can remove this flag and add the goals flag inside the method